tableslurp

```diff
@@ -53,96 +70,44 @@ class DownloadCounter(object):
 
 
 class DownloadHandler(object):
-    key = None
-    secret = None
-    token = None
-    bucket_name = None
-    owner = None
-    group = None
-    preserve = False
-    target = None
-    origin = None
-    prefix = None
-    force = False
-    name = socket.getfqdn()
-    fileset = []
-    queue = Queue()
-    num_threads = 4
-    threads = {}
-
-    def __init__(self, args):
-        self.target = args.target[0]
-        self.origin = args.origin[0]
-        self.preserve = args.preserve
-        self.key = args.aws_key
-        self.secret = args.aws_secret
-        self.region = args.aws_region
-        self.token = args.token
-        self.bucket_name = args.bucket[0]
-        self.num_threads = args.threads
-        self.force = args.force
-        if args.name:
-            self.name = args.name
-        self.prefix = '%s:%s' % (self.name, self.origin)
+    def __init__(self, key=None, secret=None, region=None, token=None, bucket_name=None,
+                 owner=None, group=None, preserve=False, target=None, prefix=None,
+                 force=False, name=socket.getfqdn(), fileset=None, num_threads=4):
```

Comment:
You're calling socket.getfqdn() twice here. Ideally you'd set it to None in the arguments, then do something like

    if name is None:
        name = socket.getfqdn()

Not really sure what was wrong with using static variables for the defaults though...

Reply:
Yeah I totally missed `socket.getfqdn()` twice...

About static variables for defaults, I can't quite remember. Something about how I had wanted to instantiate `DownloadHandler` with different parameters depending on provided arguments. I can change it back if you like?

Reply:
Yeah, to get it to work with the original, the arguments passed into `DownloadHandler` are slightly different. Some of the variables if left static will be ok, others like `target` and `prefix` I need to change it before instantiating `DownloadHandler`.